semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+from __future__ import print_function
+import sys
+from collections import namedtuple
+import os.path
+from datetime import datetime as dt
+import numpy as np
+from scipy.interpolate import CloughTocher2DInterpolator
+
+from ecl.grid import EclGrid
+from ecl.gravimetry import EclSubsidence
+from ecl.eclfile import Ecl3DKW, EclFile
+
+from ecl.util.geometry import Surface
+
+from .ots_vel_surface import OTSVelSurface
+from .ots_res_surface import OTSResSurface
+from .ots_config import build_schema
+import configsuite
+import yaml
+
+
+def extract_ots_context(configuration):
+    if configuration.eclbase is not None:
+        rstfile_path = "{}.UNRST".format(configuration.eclbase)
+        rstfile = EclFile(rstfile_path)
+        dates = [d.date() for d in rstfile.dates]
+        return dates
+    return []
+
+
+def ots_load_params(input_file):
+    config = None
+    with open(input_file, "r") as fin:
+        config = yaml.safe_load(fin)
+    config = configsuite.ConfigSuite(
+        config,
+        build_schema(),
+        deduce_required=True,
+        extract_validation_context=extract_ots_context,
+    )
+    assert config.valid
+
+    input_data = config.snapshot
+    return input_data, input_data.vintages
```

Comment:
it is a bit odd to split the return into both returning the snapshot, and a part of the snapshot. I mean, you're already returning the snapshot..

But otherwise I think it's interesting that the `vintages` turns into `vintage_pairs` on the other side. Are they vintages or vintage pairs?

Reply:
Those are vintage pairs to start with, but agree this can just output `input_data` as I did this for old-sake compatibility :)